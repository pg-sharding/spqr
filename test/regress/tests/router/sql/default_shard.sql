\c spqr-console

-- check that numeric type works
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
CREATE KEY RANGE FROM -20 ROUTE TO sh2 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM -10 ROUTE TO sh3 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 0 ROUTE TO sh4 FOR DISTRIBUTION ds1;

ALTER DISTRIBUTION ds1 ADD DEFAULT SHARD sh1;

CREATE DISTRIBUTED RELATION def_sh_xx DISTRIBUTION KEY id IN ds1;

\c regress


CREATE TABLE def_sh_xx(id INT);

INSERT INTO def_sh_xx (id) VALUES (-30);
INSERT INTO def_sh_xx (id) VALUES (-21);
INSERT INTO def_sh_xx (id) VALUES (-20);
INSERT INTO def_sh_xx (id) VALUES (-19);
INSERT INTO def_sh_xx (id) VALUES (-11);
INSERT INTO def_sh_xx (id) VALUES (-10);
INSERT INTO def_sh_xx (id) VALUES (-9);
INSERT INTO def_sh_xx (id) VALUES (0);
INSERT INTO def_sh_xx (id) VALUES (1);

DROP TABLE def_sh_xx;


\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
