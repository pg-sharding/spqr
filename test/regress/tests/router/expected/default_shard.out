\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

-- check that numeric type works
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE kr3 FROM 0 ROUTE TO sh4 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

CREATE KEY RANGE kr2  FROM -10 ROUTE TO sh3 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> -10
(1 row)

CREATE KEY RANGE kr1 FROM -20 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> -20
(1 row)

ALTER DISTRIBUTION ds1 ADD DEFAULT SHARD sh1;
  create default shard  
------------------------
 distribution id -> ds1
 shard id        -> sh1
(2 rows)

CREATE DISTRIBUTED RELATION def_sh_xx DISTRIBUTION KEY id IN ds1;
         attach table         
------------------------------
 relation name   -> def_sh_xx
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE def_sh_xx(id INT);
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
INSERT INTO def_sh_xx (id) VALUES (-30);
NOTICE: send query to shard(s) : sh1
INSERT INTO def_sh_xx (id) VALUES (-21);
NOTICE: send query to shard(s) : sh1
INSERT INTO def_sh_xx (id) VALUES (-20);
NOTICE: send query to shard(s) : sh2
INSERT INTO def_sh_xx (id) VALUES (-19);
NOTICE: send query to shard(s) : sh2
INSERT INTO def_sh_xx (id) VALUES (-11);
NOTICE: send query to shard(s) : sh2
INSERT INTO def_sh_xx (id) VALUES (-10);
NOTICE: send query to shard(s) : sh3
INSERT INTO def_sh_xx (id) VALUES (-9);
NOTICE: send query to shard(s) : sh3
INSERT INTO def_sh_xx (id) VALUES (0);
NOTICE: send query to shard(s) : sh4
INSERT INTO def_sh_xx (id) VALUES (1);
NOTICE: send query to shard(s) : sh4
SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh1 */;
NOTICE: send query to shard(s) : sh1
 id  
-----
 -30
 -21
(2 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh2 */;
NOTICE: send query to shard(s) : sh2
 id  
-----
 -20
 -19
 -11
(3 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh3 */;
NOTICE: send query to shard(s) : sh3
 id  
-----
 -10
  -9
(2 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh4 */;
NOTICE: send query to shard(s) : sh4
 id 
----
  0
  1
(2 rows)

TRUNCATE def_sh_xx;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
COPY def_sh_xx (id) FROM STDIN;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh1 */;
NOTICE: send query to shard(s) : sh1
 id  
-----
 -30
 -21
(2 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh2 */;
NOTICE: send query to shard(s) : sh2
 id  
-----
 -20
 -19
 -11
(3 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh3 */;
NOTICE: send query to shard(s) : sh3
 id  
-----
 -10
  -9
(2 rows)

SELECT * FROM def_sh_xx ORDER BY id /* __spqr__execute_on: sh4 */;
NOTICE: send query to shard(s) : sh4
 id 
----
  0
  1
(2 rows)

DROP TABLE def_sh_xx;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

